src/controllers/auth/login.ts

```diff
@@ -0,0 +1,26 @@
+import * as Interfaces from "@interfaces";
+import * as Utils from "@utils";
+
+const login: Interfaces.Middlewares.Async = async (req, res, next) => {
```

Comment:
Almost all of it will be shifted to passport when you add it. And the req.user will be populated by passport with user stuff